### DISCUSSION TITLE
using EZDXF to output PNG images, Chinese characters cannot be displayed normally

### DISCUSSION THREAD

**Discussion by JeanWang0824** [mozman/ezdxf#1143] [Q&A] [answered]
EZDXF is really useful for automating the generation of CAD drawings! When the generated DXF file is opened in AutoCAD, Chinese characters can be displayed normally. However, when using EZDXF to output PNG images, Chinese characters cannot be displayed normally. What should I do

**Comment by JeanWang0824** [comment, marked as answer]
The problem has been solved. I hope it will be helpful to future guys.
The reason is that AutoCAD replaces fonts that are not available with usable fonts, so Chinese can be displayed normally in AutoCAD, but ezdxf does not have this operation.
When using ezdxf and adding Chinese, it is necessary to specify the arialuni.ttf font, and the operating system should also install this font. When exporting PNG images, Chinese can be displayed normally.
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
doc.styles.add('myChineseStyle', font=font_path)
dim = msp.add_aligned_dim(p1=pos_mid, p2=pos_end, distance=5.0, text=name,
                              override={'dimtxsty': 'myChineseStyle', "dimtxt": 200, "dimlwd": 100})

**Comment by mozman** [comment]
see also #1032

**Reply by JeanWang0824** [reply]
hi，I did refer to the example you mentioned and solve the problem of not displaying Chinese characters in PNG images output under Windows system. Strangely, it still cannot be displayed under Kirin operating system

**Reply by JeanWang0824** [reply]
I placed arialuni.ttf in/usr/share/fonts/arialuni.ttf, executed sudo fc cache fv to update the font cache, and executed ezdxf -- fonts to see the newly added fonts. However, the output PNG image still cannot display Chinese properly under the Kirin operating system

**Reply by mozman** [reply]
I assume the DXF file still refers the font without unicode support. You also have to modify the text styles which should use `arialuni.ttf`. `Ezdxf` uses `arialuni.ttf` as replacement font  __only__ when the requested font of the text style isn't available.

**Reply by JeanWang0824** [reply]
The problem has been solved! Thank you for helping me identify the root cause of the problem, which is the missing font. I first solved the problem on the Windows system, where I actually installed arialuni.ttf twice. 
~/AppSata/Local/Microsoft/Windows/Fonts
*SystemRoot*/Fonts
 there are actually multiple fonts starting with arial. The first time, I only copied the original font files to the Kirin operating system's/usr/share/fonts and updated the font cache, but the output PNG still couldn't display Chinese characters properly. Finally, I copied all the fonts starting with arial from the Windows system to the Kirin operating system, and the problem was resolved. Although I still don't know which font worked in the end, it doesn't matter, as long as it works.Thanks again。
